docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,460 @@
+
+.. tab:: TensorFlow
+
+   - `tensorflow-onnx (tf2onnx) <https://github.com/onnx/tensorflow-onnx>`_ documentation.
+
+.. tab:: Scikit Learn
+
+   TODO
+
+Converting model frameworks to ONNX format
+-----------------------------------------------
+
+.. note::
+
+   BentoML currently only support `ONNX Runtime
+   <https://onnxruntime.ai>`_ as ONNX backend. BentoML requires either
+   ``onnxruntime>=1.9`` or ``onnxruntime-gpu>=1.9`` to be installed.
+
+.. tab:: PyTorch
+
+   First, let’s create a SuperResolution model in PyTorch.
+
+   .. code-block:: python
+
+      import torch.nn as nn
+      import torch.nn.init as init
+
+      class SuperResolutionNet(nn.Module):
+	  def __init__(self, upscale_factor, inplace=False):
+	      super(SuperResolutionNet, self).__init__()
+
+	      self.relu = nn.ReLU(inplace=inplace)
+	      self.conv1 = nn.Conv2d(1, 64, (5, 5), (1, 1), (2, 2))
+	      self.conv2 = nn.Conv2d(64, 64, (3, 3), (1, 1), (1, 1))
+	      self.conv3 = nn.Conv2d(64, 32, (3, 3), (1, 1), (1, 1))
+	      self.conv4 = nn.Conv2d(32, upscale_factor ** 2, (3, 3), (1, 1), (1, 1))
+	      self.pixel_shuffle = nn.PixelShuffle(upscale_factor)
+
+	      self._initialize_weights()
+
+	  def forward(self, x):
+	      x = self.relu(self.conv1(x))
+	      x = self.relu(self.conv2(x))
+	      x = self.relu(self.conv3(x))
+	      x = self.pixel_shuffle(self.conv4(x))
+	      return x
+
+	  def _initialize_weights(self):
+	      init.orthogonal_(self.conv1.weight, init.calculate_gain('relu'))
+	      init.orthogonal_(self.conv2.weight, init.calculate_gain('relu'))
+	      init.orthogonal_(self.conv3.weight, init.calculate_gain('relu'))
+	      init.orthogonal_(self.conv4.weight)
+
+      torch_model = SuperResolutionNet()
+
+   For this tutorial, we will download some pre-trained weights. Note
+   that this model was not trained fully for good accuracy and is used
+   here for demonstration purposes only.
+
+   .. code-block:: python
+
+      # Load pretrained model weights
+      model_url = 'https://s3.amazonaws.com/pytorch/test_data/export/superres_epoch100-44c6958e.pth'
+
+      # Initialize model with the pretrained weights
+      map_location = lambda storage, loc: storage
+      if torch.cuda.is_available():
+	  map_location = None
+      torch_model.load_state_dict(model_zoo.load_url(model_url, map_location=map_location))
+
+      # set the model to inference mode
+      torch_model.eval()
+
+
+   Exporting a model to onnx in PyTorch works via tracing or
+   scripting. In this tutorial we will export a model using
+   tracing. Note how we export the model with an input of
+   ``batch_size=1``, but then specify the first dimension as dynamic
+   in the ``dynamic_axes`` parameter in ``torch.onnx.export()``. The
+   exported model will thus accept inputs of size ``[batch_size, 1,
+   224, 224]`` where ``batch_size`` can vary among each inference.
```

Comment:
thoughts: Does this mean `bentoml.torchscript` is better used for this example?

Reply:
@aarnphm It's not about bentoml, the `torch.onnx.export` just has similar api like torchscript export

Reply:
noted.